File: java/jeromq/app/src/main/java/com/example/jeromq/MessageListenerHandler.java
package com.example.jeromq;

import android.os.Handler;
import android.os.Message;

public class MessageListenerHandler extends Handler {

	public MessageListenerHandler(MessageListener messageListener, String payloadKey) {
		_messageListener = messageListener;
		_payloadKey = payloadKey;
	}

	@Override
	public void handleMessage(Message msg) {
		_messageListener.messageReceived(msg.getData().getString(_payloadKey));
	}

	private final MessageListener _messageListener;
	private final String _payloadKey;
}

File: kotlin/jeromq/jeromq/app/src/main/java/example/kotlin/jeromq/MessageListenerHandler.kt
package example.kotlin.jeromq

import android.os.Handler
import android.os.Message

class MessageListenerHandler(
	private val listener: MessageListener,
	private val payloadKey: String
) : Handler() {

	override fun handleMessage(msg: Message?) {
		listener.messageReceived(msg!!.data.getString(payloadKey))
	}
}